client/src/components/ProductDetails/ImageGallery.jsx

```diff
@@ -13,23 +13,66 @@ function ImageGallery ({productStylesArray, currentStyleID}) {
 
   const [currentImage, updateMainImage] = useState(imageGallery[0]);
 
+  const [currentMainImageIndex, updateMainImageIndex] = useState(0);
+
   const setThumbnailClass = (image) => {
     return image === currentImage ? 'imageThumbnailMain' : 'imageThumbnail';
   };
 
 
+  const showNextImage = ()=>{
+    if(currentMainImageIndex !== imageGallery.length-1) {
+      updateMainImageIndex(currentMainImageIndex+1);
+      const newImg = imageGallery.filter((image)=> imageGallery.indexOf(image) === currentMainImageIndex+1);
+      updateMainImage(newImg);
+      // setThumbnailClass(imageGallery.indexOf(currentMainImageIndex+1));
+    }
+  };
+
+  const showPreviousImage = ()=>{
+    if(currentMainImageIndex > 0) {
+      updateMainImageIndex(currentMainImageIndex-1);
+      const newImg = imageGallery.filter((image)=> imageGallery.indexOf(image) === currentMainImageIndex-1);
+      updateMainImage(newImg);
+      // setThumbnailClass(imageGallery.indexOf(currentMainImageIndex-1));
+    }
+  };
+
+
   return (
-    <div clasName="container">
-      <img className="mainImage" src={currentImage}/>
+    <div className="container main-image-container">
+      <img className="mainImage" src={currentImage} />
+      <input
+        type="image"
+        className="previousButton"
+        src="https://img.icons8.com/ios-glyphs/50/000000/long-arrow-left.png"
+        onClick= { () => showPreviousImage()}/>
+      <input
+        type="image"
+        className="nextButton"
+        src="https://img.icons8.com/ios-glyphs/50/000000/long-arrow-right.png"
+        onClick= { () => showNextImage() }/>
+
+
       {imageGallery.map((image) =>
         // eslint-disable-next-line react/jsx-key
-        <div clasName="row">
+        <div className="row">
```

Comment:
I think you can pass in a key attribute to the div and define the key using string literals i.e. `some-class-name ${index}` and that way you won't need to disable eslint here